pandas/core/series.py

```diff
@@ -4331,6 +4334,64 @@ def between(self, left, right, inclusive=True) -> "Series":
 
         return lmask & rmask
 
+    # ----------------------------------------------------------------------
+    # Convert to types that support pd.NA
+
+    def _as_nullable_dtype(self: ABCSeries, keep_integer: bool = False) -> ABCSeries:
+        """
+        Handle one Series
+
+        Rules:
+        If an object, see if we can infer string, boolean or integer, otherwise leave
+        alone
+        If an integer and not an extension type, convert to the extension Int type
+        If numeric, see if we can infer integer, otherwise try to use astype() to make
+        it integer.
+
+        """
+        dtype = self.dtype
+        new_dtype = dtype
+        changeit = True
+        result = self
+        target_int_dtype = "Int64"
+
+        if is_object_dtype(dtype):
```

Comment:
I would simply start by lib.infer_dtype upfront; for a known type it is O(1)

Reply:
done